src/Domain/FunctionsOfTime/FunctionOfTimeHelpers.hpp

```diff
@@ -0,0 +1,50 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// \brief Helper struct and functions for FunctionsOfTime
+
+#pragma once
+
+#include <array>
+#include <limits>
+#include <pup.h>
+
+#include "DataStructures/DataVector.hpp"
+#include "Utilities/Gsl.hpp"
+
+/// \brief Stores info at a particlar time about a function and its derivatives
+///
+/// \details `MaxDerivPlusOne` template parameter is intended to be related to
+/// the maximum derivative that a FunctionOfTime is supposed to store. For a
+/// generic PiecewisePolynomial this will be the MaxDeriv + 1 (because we store
+/// the function as well as its derivatives)
+template <size_t MaxDerivPlusOne>
+struct StoredInfo {
+  double time{std::numeric_limits<double>::signaling_NaN()};
+  std::array<DataVector, MaxDerivPlusOne> stored_quantities;
+
+  StoredInfo() = default;
+
+  StoredInfo(double t, std::array<DataVector, MaxDerivPlusOne> init_quantities,
+             bool StoringCoefs = true) noexcept;
```

Comment:
Please document what `StoringCoefs` means.  (And is there a better name than `StoringCoefs`? Seems strange to have a gerund here.)

Also, if `StoringCoefs` is what I think it is (i.e. always true for PiecewisePolynomial and always false for QuatFvT), should it be a bool template parameter instead of a constructor argument?